api/provider/bitbucket.go

```diff
@@ -0,0 +1,63 @@
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/netlify/gotrue/conf"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/bitbucket"
+)
+
+// Bitbucket
+
+type bitbucketProvider struct {
+	*oauth2.Config
+}
+
+func NewBitbucketProvider(ext conf.ExternalConfiguration) Provider {
+	return &bitbucketProvider{
+		&oauth2.Config{
+			ClientID:     ext.Key,
+			ClientSecret: ext.Secret,
+			Endpoint:     bitbucket.Endpoint,
+		},
+	}
+}
+
+func (g bitbucketProvider) GetOAuthToken(ctx context.Context, code string) (*oauth2.Token, error) {
+	return g.Exchange(ctx, code)
+}
+
+func (g bitbucketProvider) GetUserEmail(ctx context.Context, tok *oauth2.Token) (string, error) {
```

Comment:
Okay, I think I get what you're talking about now -- sorry for the confusion.

I've added a makeRequest function like the one you've mentioned above for all GET requests to external providers.